.github/workflows/shared-create-release.yml

```
@@ -0,0 +1,216 @@
+            exit 1
+          fi
+          if [ $(git tag -l "$VERSION") ]; then
+             echo $VERSION already exists
+            exit 1
+          fi
+
+      - name: Determine Next Version
+        run: |
+
+          # joins an array with a delimiter
+          function join_by { local IFS="$1"; shift; echo "$*"; }
+
+          # create a list of tags that are unrelated to the current release
+          ALL_TAGS=$(git tag -l)
+          temp_array=()
+          for value in ${ALL_TAGS[@]}
+          do
+              if ! [[ $value =~ ${{ inputs.main-tag-regex }} ]]; then
+                temp_array+=($value)
+              fi
+          done
+          UNRELATED_TAGS=$(join_by , "${temp_array[@]}")
+          unset temp_array
+          echo UNRELATED_TAGS=$UNRELATED_TAGS
+
+          # set variables
+          FIRST="--exclude-tags "
+          SECOND=$UNRELATED_TAGS
+          EXCLUDE_ARG=$FIRST$SECOND
+          echo EXCLUDE_ARG=$EXCLUDE_ARG
+          echo EXCLUDE_ARG=$EXCLUDE_ARG >> $GITHUB_ENV
+
+      - name: Prepare repository
+        run: |
+          GITHUB_REPOSITORY_USER=$( echo $GITHUB_REPOSITORY | awk -F'/' '{print $1}') 
+          GITHUB_REPOSITORY_PROJECT=$( echo $GITHUB_REPOSITORY | awk -F'/' '{print $2}') 
+
+          echo GITHUB_REPOSITORY_USER=$GITHUB_REPOSITORY_USER
+          echo GITHUB_REPOSITORY_PROJECT=$GITHUB_REPOSITORY_PROJECT
+
+          echo GITHUB_REPOSITORY_USER=$GITHUB_REPOSITORY_USER >> $GITHUB_ENV
+          echo GITHUB_REPOSITORY_PROJECT=$GITHUB_REPOSITORY_PROJECT >> $GITHUB_ENV
+
+      - name: Run github-changelog-generator
+        uses: docker://githubchangeloggenerator/github-changelog-generator
+        with:
+          args: >
+            --output ${{ inputs.changelog-path }}
+            --config-file ${{ inputs.changelog-config-path }}
+            --user ${{ env.GITHUB_REPOSITORY_USER }}
+            --project ${{ env.GITHUB_REPOSITORY_PROJECT }} 
+            --token ${{ secrets.GITHUB_TOKEN }} 
+            --future-release ${{ inputs.tag-prefix }}${{ inputs.tag-base }} 
+            ${{ env.EXCLUDE_ARG }}
+
+      - name: Log ${{ inputs.changelog-path }}
+        run: cat ${{ inputs.changelog-path }}
+
+      - name: Import GPG key
+        id: import-gpg
+        uses: crazy-max/ghaction-import-gpg@v4
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+          git_tag_gpgsign: true
+
+      - name: Commit changelog
+        run: |
+          git add .
+          if [[ $(git diff --stat --staged) == '' ]]; then
+            echo 'repository unmodified'
+            exit 1
+          fi
+          git commit -m "generate changelog"
+
+      - uses: actions-rs/toolchain@v1
+        if: ${{inputs.release-target == 'rust' || inputs.release-target == 'wasm'}}
+        with:
+          toolchain: stable
```

Needs to override the toolchain to be consistent.

I think we'll always need the Rust toolchain, does it need the conditional on release-target?

```suggestion
      - name: Install Rust toolchain
        uses: actions-rs/toolchain@v1
        if: ${{inputs.release-target == 'rust' || inputs.release-target == 'wasm'}}
        with:
          toolchain: stable
          profile: minimal
          override: true
```